codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/Instantiator.kt

```diff
@@ -222,14 +229,23 @@ class Instantiator(
      */
     private fun renderUnion(writer: RustWriter, shape: UnionShape, data: ObjectNode, ctx: Ctx) {
         val unionSymbol = symbolProvider.toSymbol(shape)
-        check(data.members.size == 1)
-        val variant = data.members.iterator().next()
-        val memberName = variant.key.value
+
+        val variant = if (ctx.defaultsForRequiredFields && data.members.isEmpty()) {
```

Comment:
It'd be nice to destructure here, maybe something like this?
```suggestion
        val (memberName, defaultValue) = if (ctx.defaultsForRequiredFields && data.members.isEmpty()) {
```